src/chat.rs

```diff
@@ -2923,10 +2994,14 @@ pub async fn forward_msgs(context: &Context, msg_ids: &[MsgId], chat_id: ChatId)
             msg.subject = "".to_string();
 
             let new_msg_id: MsgId;
-            if msg.state == MessageState::OutPreparing {
+            if msg.state == MessageState::OutDraft {
+                bail!("cannot forward drafts.");
```

Comment:
Seems better to do this check right after loading the message, before preparing the `param` which is then thrown away.